sdk/rust/src/test.rs

```diff
@@ -8,52 +8,42 @@ mod tests {
     const PLAINT_TEXT: &str = "Intel test";
 
     #[tokio::test]
-    async fn test_asymmetrickey_generate_key() {
+    async fn test_generate_key() {
         let mut client = EHSMClient::new();
 
-        let keyspecs = vec!["EH_RSA_3072", "EH_RSA_4096", "EH_RSA_2048", "EH_SM2", "EH_EC_P224", "EH_EC_P256", "EH_EC_P384", "EH_EC_P521"];
-        let padding_modes = vec![Some("EH_PAD_RSA_PKCS1_OAEP"), Some("EH_PAD_RSA_PKCS1"), None];
-        let digest_modes = vec![Some("EH_SHA_2_224"), Some("EH_SHA_2_256"), Some("EH_SHA_2_384"), Some("EH_SHA_2_512")];
+        let keyspecs = vec!["EH_RSA_2048", "EH_RSA_3072", "EH_RSA_4096", "EH_SM2", "EH_EC_P224", "EH_EC_P256", "EH_EC_P384", "EH_EC_P521", "EH_AES_GCM_128", "EH_AES_GCM_192", "EH_AES_GCM_256", "EH_SM4_CTR", "EH_SM4_CBC"];
+        let keyusage = vec!["EH_KEYUSAGE_ENCRYPT_DECRYPT", "EH_KEYUSAGE_SIGN_VERIFY"]
         for keyspec in keyspecs.iter() {
```

Comment:
EH_KEYUSAGE_SIGN_VERIFY can't use for symmetric key, and does we have check in the low level implementation?

Reply:
if not, then we need to add it @whh39

Reply:
I add it on https://github.com/intel/ehsm/pull/298